aspnetcore/release-notes/aspnetcore-10/includes/jsonPatch.md

```diff
@@ -0,0 +1,240 @@
+* Represents a sequence of operations (for example, Add, Remove, Replace, Move, Copy, Test) that can be applied to modify a JSON document.
+
+In web apps, JSON Patch is commonly used in a PATCH operation to perform partial updates of a resource. Rather than sending the entire resource for an update, clients can send a JSON Patch document containing only the changes. Patching reduces payload size and improves efficiency.
+
+[RFC 6902]: https://tools.ietf.org/html/rfc6902
+
+This release introduces a new implementation of [`JsonPatch`](/dotnet/api/system.text.json based on `System.Text.Json` serialization. This feature:
+
+* Aligns with modern .NET practices by leveraging the `System.Text.Json` library, which is optimized for .NET.
+* Provides improved performance and reduced memory usage compared to the legacy `Newtonsoft.Json`-based implementation.
+
+The following benchmarks compare the performance of the new `System.Text.Json` implementation with the legacy `Newtonsoft.Json` implementation:
+
+| Scenario                   | Implementation         | Mean       | Allocated Memory |
+|----------------------------|------------------------|------------|------------------|
+| **Application Benchmarks** | Newtonsoft.JsonPatch   | 271.924 µs | 25 KB            |
+|                            | System.Text.JsonPatch  | 1.584 µs   | 3 KB             |
+| **Deserialization Benchmarks** | Newtonsoft.JsonPatch | 19.261 µs  | 43 KB            |
+|                            | System.Text.JsonPatch  | 7.917 µs   | 7 KB             |
+
+These benchmarks highlight significant performance gains and reduced memory usage with the new implementation.
+
+Notes:
+ * The new implementation isn't a drop-in replacement for the legacy implementation. In particular:
+   * The new implementation doesn't support dynamic types, for example, [`ExpandoObject`](/dotnet/api/system.dynamic.expandoobject.
+ * The JSON Patch standard has ***inherent security risks***. Since these risks are inherent to the JSON Patch standard, the new implementation ***doesn't attempt to mitigate inherent security risks***. It's the responsibility of the developer to ensure that the JSON Patch document is safe to apply to the target object. For more information, see the [Mitigating Security Risks](#mitigating-security-risks) section.
+
+#### Usage
+
+To enable JSON Patch support with `System.Text.Json`, install the [`Microsoft.AspNetCore.JsonPatch.SystemTextJson`](https://www.nuget.org/packages/Microsoft.AspNetCore.JsonPatch/) NuGet package.
+
+```sh
+dotnet add package Microsoft.AspNetCore.JsonPatch.SystemTextJson --prerelease
+```
+
+This package provides a `JsonPatchDocument<T>` class to represent a JSON Patch document for objects of type `T` and custom logic for serializing and deserializing JSON Patch documents using `System.Text.Json`. The key method of the `JsonPatchDocument<T>` class is `ApplyTo`, which applies the patch operations to a target object of type `T`.
```

Comment:
Once the reference docs for Preview 4 are published, I think we should change the "code-style" references to .NET types to link to the ref docs.